src/dispatching/dispatcher.rs

```diff
@@ -438,6 +440,67 @@ where
         }
     }
 
+    fn hint_allowed_updates<E>(&self, listener: &mut impl UpdateListener<E>) {
+        let mut allowed = self
+            .messages_queue
+            .as_ref()
+            .map(|_| AllowedUpdate::Message)
+            .into_iter()
+            .chain(
+                self.edited_messages_queue
+                    .as_ref()
+                    .map(|_| AllowedUpdate::EditedMessage)
+                    .into_iter(),
+            )
```

Comment:
Consider segregating such pieces of code into a separate function.

Reply:
Done